frontend-react/src/components/Admin/EditReceiverSettings.tsx

```diff
@@ -343,36 +345,47 @@ const EditReceiverSettingsForm: React.FC<EditReceiverSettingsFormProps> = ({
     );
 };
 
-type Props = {
-    orgname: string;
-    receivername: string;
+type EditReceiverSettingsProps = {
+    orgName: string;
+    receiverName: string;
     action: "edit" | "clone";
 };
 
-export function EditReceiverSettings({ match }: RouteComponentProps<Props>) {
-    const orgname = match?.params?.orgname || "";
-    const receivername = match?.params?.receivername || "";
-    const action = match?.params?.action || "";
+export function EditReceiverSettings() {
+    const { orgName, receiverName, action } =
+        useParams<EditReceiverSettingsProps>();
+    /* useParams now returns possible undefined. This will error up to a boundary
+     * if the url param is undefined */
+    if (
```

Comment:
Might behoove us later to refactor this into some kind of `useValidParams` hook that throws whenever expected `!=` received params (i.e. middleware undefined check)